Isaacs/remove deps no longer used

isaacs opened the pull request:
Various cleanups, found a gnarly bundle deps install bug.

isaacs committed:
remove qw dependency

Arrays of string literals are fine, and we're only using it in one place.

ethomson committed:
GitHub Actions: add a win32 bash workflow

Update the matrix to have two Windows workflows: one running PowerShell
and one running bash.

ruyadorno committed:
chore: removed .travis.yml file

PR-URL: https://github.com/npm/cli/pull/1583
Credit: @ethomson
Close: #1583
Reviewed-by: @isaacs

isaacs committed:
npx: add install prompt, handle options correctly

- handle previous npx options that are still possible to be handled, and
  print a warning if any deprecated/removed options are used.
- expand shorthands properly in npx command line.
- take existing npm options into account when determining placement of
  the -- argument.
- document changes from previous versions of npx.

PR-URL: https://github.com/npm/cli/pull/1596
Credit: @isaacs
Close: #1596
Reviewed-by: @ruyadorno

isaacs committed:
remove unused production deps

isaacs committed:
gitignore package-lock.json files in node_modules

isaacs committed:
ignore coverage and packed cli versions

isaacs committed:
Reset dep install. Discover strange bundle dep bug

The `tap` package bundles `import-jsx`, which in turn bundles
`@babel/core` and a bunch of other stuff.

However, even though `@babel/core` is in the bundle, it's getting placed
in the tree at the root.  This is a mistake.

Because the node is placed, and not given the `extraneous` flag, in the
ideal tree building process, when reified, it goes ahead and puts it in
place.  When reading the actual tree, we see a valid hidden lockfile,
which contains the module, and it doens't have the extraneous flag.  So
everything seems fine.

Remove the hidden lockfile, and the problem presents itself.

Corrected here through some arduous and painstaking manual installs, so
at least we have the tree in the state it *should* be for now.  But this
is a bug in arborist that must be fixed asap.

Re: https://github.com/npm/cli/issues/1597

isaacs committed:
Set stdioString: true on install scripts

isaacs committed:
update: depth defaults to 0 now, not Infinity

isaacs committed:
@npmcli/arborist@0.0.11

Fix #1597

isaacs committed:
completion: code cleanup. still pending full unit tests

ruyadorno reviewed (approved):
LGTM 👍 will wait for #1596

isaacs committed:
remove unused lifecycle util

isaacs committed:
bin-links@2.1.3 @npmcli/arborist@0.0.12

isaacs committed:
Get 'make htmldocs' working

There's still a task pending to get it actually up to date with
non-conflicting peer deps and everything updated to remove deprecated
and outdated packages.  But at least it builds at all now.

isaacs committed:
test: verify all commands have usage and completion

isaacs committed:
refactor npm init, use npm exec instead of libnpx

isaacs committed:
Fix lingering figgy puddy in login

isaacs committed:
refactor and lint commands, make consistent

Still pending test coverage for most of these, but wanted to give them a
clean sweep to get the "load-all-commands" tests passing.

The following changes are in here:

- All commands now have a `completion()` method and a usage string that
  uses the same `usage` util consistently.
- The `silent` argument to many commands has been removed.
- All commands use the `cmd = ${cmd}(args).then(() => cb()).catch(cb)`
  pattern consistently.

Full test coverage for all commands is still lacking, and will have to
be done prior to the GA v7 release.

isaacs committed:
Actions: remove v6 and v8, add v14

Accidentally landed this in the v7 branch, didn't realize it was based on v6.

isaacs committed:
test: enforce full coverage

Also, get Windows up to full coverage.

ruyadorno commented:
✅ merged
